### ISSUE TITLE
For simple arguments, values `0` and `false` are replaced by empty string

### ISSUE THREAD

**Issue by arnebit** [ember-intl/ember-intl#512]
### Environment
- Ember Version: 2.12.2
- Ember CLI Version: 2.12.1
- Ember Intl Version: 2.27.0

### Steps to Reproduce
```
intl.formatMessage('foo {x} bar',{x:true})  // "foo true bar"
intl.formatMessage('foo {x} bar',{x:false})  // "foo  bar" <- inconsistent, no "false"

intl.formatMessage('foo {x} bar',{x:1})  // "foo 1 bar"
intl.formatMessage('foo {x} bar',{x:0})  // "foo  bar" <- no "0"
```

Is this behavior intentional or just the result of some too eager `value || ''` in some place where `value===undefined || value===null ? '' : value` would have fit better?

I am not sure I am navigating the ICU spec correctly, but [one linked (java) document](http://icu-project.org/apiref/icu4j/com/ibm/icu/text/MessageFormat.html) says 

> An argument might not specify any format type. In this case, a Number value is formatted with a default (for the locale) NumberFormat,

which, if I recall Java behaviour correctly, would at least give '0' for value `0`.

**Comment by jasonmit** [comment]
https://github.com/yahoo/intl-messageformat/pull/156 I'd love to get this merged.  It addresses the issue - may end up forking intl-messageformat if it's not addressed soon.

**Comment by jasonmit** [comment]
Also - see this comment on how to handle this: https://github.com/yahoo/intl-messageformat/issues/88#issuecomment-68831305

**Comment by jasonmit** [comment]
This is now resolved within intl-messageformat and 2.29.0 contains the dependency upgrade.
